modules/imgproc/src/color_hsv.simd.hpp

```diff
@@ -955,12 +955,11 @@ struct HLS2RGB_f
                 float p1 = 2*l - p2;
 
                 h *= hscale;
-                if( h < 0 )
-                    do h += 6; while( h < 0 );
-                else if( h >= 6 )
-                    do h -= 6; while( h >= 6 );
+                // We need both loops to clamp (e.g. for h == -1e-40).
+                do h += 6; while( h < 0 );
+                do h -= 6; while( h >= 6 );
```

Comment:
They were in `while..do` form (due to `if` condition before)

Check code path for `h = 1` without and with patch.

Reply:
good catch, fixed.